frontend/src/components/scan/utils/RuntimeConfigurationModal.jsx

```diff
@@ -7,123 +7,151 @@ import { ContentSection, CustomJsonInput } from "@certego/certego-ui";
-      ),
-    }),
-    [formik.values.analyzers, formik.values.connectors, analyzers, connectors],
-  );
-
-  console.debug("RuntimeConfigurationModal - combinedParamsMap:");
-  console.debug(combinedParamsMap);
-
-  // Iterate each plugin and for each param extract the value
-  const defaultNameParamsMap = React.useMemo(
-    () =>
-      Object.entries(combinedParamsMap).reduce(
-        (generalConfig, [pluginName, pluginParams]) => ({
-          ...generalConfig,
-          // For each param (dict) extract the value of the "value" key
-          [pluginName]: Object.entries(pluginParams).reduce(
-            (singlePluginConfig, [paramName, { value: paramValue }]) => ({
-              ...singlePluginConfig,
-              [paramName]: paramValue,
-            }),
-            {},
-          ),
-        }),
-        {},
-      ),
-    [combinedParamsMap],
-  );
+    };
+  });
 
-  console.debug("RuntimeConfigurationModal - defaultNameParamsMap:");
-  console.debug(defaultNameParamsMap);
+  console.debug("RuntimeConfigurationModal - selectedPluginsParams:");
+  console.debug(selectedPluginsParams);
 
   /* this is the dict shown when the modal is open: load the default params and the previous saved config
     (in case the user update the config, save and close and reopen the modal)
-
-    IMPORTANT: We want to group the plugins in the categories (analyzers, etc...), it is more difficult to handle it in every variable
-    so we use it only when shown to the user (UI), we need to rembeber in case we edite the params more than once we need to 
-    load the data from the structure with the categories.
+    We want to show data in this format:
+    {
+      pluginType: {
+        pluginName: {
+          paramName: paramValue,
+        },
+      },
+    }
   */
-  const editableConfig = React.useMemo(() => {
-    const config = {
-      ...defaultNameParamsMap,
-      ...(formik.values.runtime_configuration.analyzers || {}), // previous values of analyzers (groupped by the previous editing) if present
-      ...(formik.values.runtime_configuration.connectors || {}), // previous values of connectors (groupped by the previous editing) if present
+  Object.keys(selectedPluginsInFormik).forEach((pluginType) => {
+    // for each plugin extract name and default params
+    Object.entries(selectedPluginsParams[pluginType]).forEach(
+      ([pluginName, pluginParams]) => {
+        // add empty dict in editableConfig for plugin that have not params
+        editableConfig[pluginType] = {
+          ...editableConfig[pluginType],
+          [pluginName]: {},
+        };
+        // for each param (dict) extract the value of the "value" key
+        Object.entries(pluginParams).forEach(
+          ([paramName, { value: paramValue }]) => {
+            editableConfig[pluginType][pluginName] = {
+              ...editableConfig[pluginType][pluginName],
+              [paramName]: paramValue,
+            };
+          },
+        );
+      },
+    );
```

Comment:
I didn't try, but I think this code could be simplified, ex:
```
editableConfig[pluginType] = {
          ...editableConfig[pluginType],
          [pluginName]: {},
        };
```
This shouldn't be required anymore: it creates a new dict each time and with unpack syntax (...) put in the new dict the old data in this way it doesn't lost the previous iterations results, but with for each I think we could move the initialization before the loop and inside the loop add only the new key.